### DISCUSSION TITLE
birdnet-go AMD64 version continued?

### DISCUSSION THREAD

**Discussion by unosonic** [tphakala/birdnet-go#53] [Q&A]
hi @tphakala, do you have plans to develop the linux amd64 version ofbirdnet-go any further?

**Comment by tphakala** [comment]
Yes absolutely, code base compiles on amd64 Linux just fine, I just haven't produced compiled binary releases lately as I have been doing it manually. Once I have time I will create a github actions workflow to automate binary builds, Windows build was automated yesterday.

Docker container image is already built by github actions and it is available for arm64 and amd64.
https://github.com/tphakala/birdnet-go/pkgs/container/birdnet-go

Docker image will be preferred deployment model going forward.

**Comment by bbrodriguez** [comment]
I struggled to get this running on x86/64 (or amd64) via the :latest tag. It also would not build via the Dockerfile for me:

```
=> [birdnet build 1/2] COPY . BirdNET-Go                                                                                                                                                                                                               26.2s
 => ERROR [birdnet build 2/2] RUN --mount=type=cache,target=/go/pkg/mod     --mount=type=cache,target=/root/.cache/go-build     cd BirdNET-Go && make TARGETPLATFORM=linux/amd64                                                                         0.7s
------
 > [birdnet build 2/2] RUN --mount=type=cache,target=/go/pkg/mod     --mount=type=cache,target=/root/.cache/go-build     cd BirdNET-Go && make TARGETPLATFORM=linux/amd64:
0.568 make: *** No targets specified and no makefile found.  Stop.
------
failed to solve: process "/bin/sh -c cd BirdNET-Go && make TARGETPLATFORM=${TARGETPLATFORM}" did not complete successfully: exit code: 2

```
I did get it up and running via this image though:

```
docker pull ghcr.io/tphakala/birdnet-go:eda3be40777488c1a4656322f89868720d65247c@sha256:d7fcd03afe34c3babaa2fc53779de1941f9b228644a506f91852a93379462f93
```

**Reply by isZumpo** [reply]
The :latest version of the birdnet-go docker is currently not working with x86 because I messed up the build script a while back... However, it has since been fixed and should be available in the next release. Otherwise, releases from the main branch could be used, using the :dev tag instead of :latest.

 **ghcr.io/tphakala/birdnet-go:dev**


I don't think there should be any issues building the docker image itself locally. What commands are you executing in order to build it currently?

**Reply by tphakala** [reply]
I will create a new release this weekend so new container images will get built. I have done siginificant changes and improvements to audio clip cleanup code.

**Reply by bbrodriguez** [reply]
> The :latest version of the birdnet-go docker is currently not working with x86 because I messed up the build script a while back... However, it has since been fixed and should be available in the next release. Otherwise, releases from the main branch could be used, using the :dev tag instead of :latest.
> 
> **ghcr.io/tphakala/birdnet-go:dev**
> 
> I don't think there should be any issues building the docker image itself locally. What commands are you executing in order to build it currently?

I git cloned bidnet-go and then used a build directive (pointing at the birdnet-go/Dockerfile) in docker compose to try and create it locally.

But like I said - I got it up and running using a different image. I'll revert to :latest once its rebuilt.

**Reply by isZumpo** [reply]
Can you try once to build it with just the normal **docker build .** command?

**Reply by bbrodriguez** [reply]
OK - so that worked. Weird.

**Reply by isZumpo** [reply]
It is probably related to the relative path you have to the dockerfile when using compose. All copys etc assumes that the workspace is in the repo root.
